### ISSUE TITLE
Add oversampling/augmentation to mitigate class imbalance

### ISSUE THREAD

**Issue by bdzyubak** [bdzyubak/torch-control#23]
The dermaMNIST dataset has a substantial class imbalance issue. This causes poor generalization - 0.75 accuracy in validation with >0.95 in training. Code needs to be added to upsample low-data classes, and augment training data to improve generalizability. 

Related to the dermaMNIST accuracy. Once done, post performance results with improved validation accuracy here. 
https://github.com/bdzyubak/torch-control/issues/20
